portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,30 +35,41 @@
   @Override
   public void init() {
     messages = new ArrayList<>();
-    messages.add("Once upon a time...");
-    messages.add("Twice upon two times...");
-    messages.add("Thrice upon three times...");
   }
 
+  /** Gets the messages stored in the messages variable in JSON format */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
     response.getWriter().println(convertToJson(messages));
   }
 
+  /** Posts a message retrieved from the form input adding it to the messages variable */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Get the message from the form and add it to the array */
+    String message = request.getParameter("text-input");
+    messages.add(message);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
   /**
    * Converts an array into a JSON string using manual String concatentation.
+   * Keys will be a String called 'string' plus the number of the String.
+   * Values will be a String containing a message (user comment).
    */
   private String convertToJson(ArrayList messages) {
     String json = "{";
-    json += "\"string1\": ";
```

Comment:
Think of your keys as variables, and as such, perhaps name it something more descriptive than 'stringx'.

Reply:
Noted! Called it 'messagex' instead.